ansible_ai_connect/users/tests/test_auth.py

```diff
@@ -112,6 +115,50 @@ def test_missing_values(self):
         self.assertFalse(data["aap_system_auditor"])
         self.assertFalse(data["aap_superuser"])
 
+    def test_get_me_endpoint_controller(self):
+        authentication = AAPOAuth2()
+        api_url = "http://controller.test/api"
+        self.assertEqual(
+            "http://controller.test/api/v2/me/", authentication.get_me_endpoint(api_url)
+        )
+
+    def test_get_me_endpoint_controller_ended(self):
+        authentication = AAPOAuth2()
+        api_url = "http://controller.test/api/"
+        self.assertEqual(
+            "http://controller.test/api/v2/me/", authentication.get_me_endpoint(api_url)
+        )
+
+    def test_get_me_endpoint_gateway(self):
+        authentication = AAPOAuth2()
+        api_url = "http://controller.test"
+        self.assertEqual(
+            "http://controller.test/api/gateway/v1/me/", authentication.get_me_endpoint(api_url)
+        )
+
+    def test_get_config_endpoint_controller(self):
+        authentication = AAPOAuth2()
+        api_url = "http://controller.test/api"
+        self.assertEqual(
+            "http://controller.test/api/v2/config/", authentication.get_config_endpoint(api_url)
+        )
+
+    def test_get_config_endpoint_gateway(self):
+        authentication = AAPOAuth2()
+        api_url = "http://controller.test"
+        self.assertEqual(
+            "http://controller.test/api/controller/v2/config/",
+            authentication.get_config_endpoint(api_url),
+        )
+
+    def test_get_config_endpoint_gateway_ended(self):
+        authentication = AAPOAuth2()
+        api_url = "http://controller.test"
```

Comment:
you meant to have trailing `/` here?

Reply:
By default `/` shouldn't be there at all (it is the same now for 2.4, but no `/` after `/api`), but I added rstrip in the `get_config_endpoint` and `get_me_endpoint` just in case.

Reply:
I am trying to understand the differences of this test case vs that in L146

Reply:
Since I check `endswith("/api")` then ending API url with `/` can affect the flow and provide Gateway API instead of expected Controller API.

So if user by mistake put `/` at the end of the API url, it shouldn't break this code. Maybe it will break something in another place but at least not here. 

And there are test cases for it to check the logic in production.

Reply:
Sorry @jameswnl, great catch, fixed!